File: 2021-1/HW01/hw0102.c
#include <stdio.h> 
#include <math.h>            
#include <ctype.h>    
#include <stdlib.h>
#include <assert.h>
#include <stdarg.h>
#include <limits.h>
#include <stdio.h>
#include <string.h>
#include <stdint.h>	
#include <time.h>

int get_max(int, ...);
int get_digits(int);
int check(int);
void split(int, int);
void split_ans(int, int);

int main()
{
    int a = 0, b = 0;
    printf("Please enter the first  number: ");
    scanf("%d", &a);
    check(a);
    printf("Please enter the second number: ");
    scanf("%d", &b);
    check(b);
    int c = a+b;

    int len = 0;
    len = get_max( 2, get_digits(a), get_digits(b) );

    printf("   "); split(a, len);
    printf("+) "); split(b, len);
    for(int i = 1; i < len; ++i) {
        printf("---");
    }
    printf("-\n ");
    split_ans(c, len);
    
    return 0;
}
void split_ans(int n, int len) {
    if( get_digits(n) == len ) {
        printf("   "); split(n, len);
    }
    else if( get_digits(n) == len+1 ) {
        split(n, len+1);
    }
    else {
        printf("ERROR\n");
    }

    return;
}
void split(int n, int len) {
    for(int i = len; i > get_digits(n); --i) {
        printf("  ");
    }
    int ten = 1;
    for(int i = 1; i < get_digits(n); ++i) {
        ten *= 10;
    }
    for(int i = get_digits(n); i > 0; --i) {
        printf("%d ", n/ten);
        n = n%ten;
        ten /= 10;
    }
    printf("\n");

    return;
}
int get_max(int n, ...) {
    int max = 0;

    va_list ptr;
    va_start(ptr, n);

    for(int i = 0; i < n; ++i) {
        int tmp = va_arg(ptr, int);
        max = tmp > max ? tmp : max;
    }

    va_end(ptr);

    return max;
}
int get_digits(int num) {
    int ret = 0;
    while(num > 0) {
        num /= 10;
        ret++;
    }
    return ret;
}
int check(int c) {
    if(c <= 0)
    {
        printf("Error\n");
    }
    return 0;
}

File: 2021-1/HW01/hw0102.py
x = 123;
print("Please enter the first  number: ");
a = int(input());
print("Please enter the second number: ");
b = int(input());
c = a+b

def get_digits(num):
    cnt = 0
    while(num != 0):
        num = int(num/10)
        cnt = cnt+1
    return cnt

len = 0;
if(a >= b):
    len = get_digits(a)
else:
    len = get_digits(b)

print(len)
print(get_digits(b))

def split(num, len):
    dig = get_digits(num)
    for i in range(len-dig):
        print(" ", end = " ")
    for i in range(dig):
        print(int(num/(10**(dig-1-i))), end = " ")
        num %= 10**(dig-1-i)

print("\n  ", end = " ")
split(a, len)
print("\n+)", end = " ")
split(b, len)
print(" ")
for i in range(len):
    print("--", end = "-")
print(" ")
if( get_digits(c) == len ):
    print("  ", end = " ")
    split(c, len)
else:
    print(" ", end = " ")
    split(c, len+1)
print(" ")